tomb-common/src/utils/serialize.rs

```diff
@@ -1,15 +1,18 @@
-use crate::types::pipeline::Manifest;
+use crate::types::blockstore::car::carv2::carv2blockstore::CarV2BlockStore;
 use anyhow::Result;
 use rand::thread_rng;
-use std::{collections::HashMap, rc::Rc};
+use std::rc::Rc;
 use wnfs::{
     common::{AsyncSerialize, BlockStore, HashOutput},
     libipld::{serde as ipld_serde, Cid, Ipld},
     private::{PrivateDirectory, PrivateForest, PrivateNode, PrivateRef, TemporalKey},
 };
 
 /// Store a given PrivateForest in a given Store
-pub async fn store_forest(forest: &Rc<PrivateForest>, store: &impl BlockStore) -> Result<Cid> {
+pub(crate) async fn store_forest(
+    forest: &Rc<PrivateForest>,
+    store: &impl BlockStore,
+) -> Result<Cid> {
```

Comment:
This pattern of creating a function for each type isn't ideal. Is there instead a trait that could be used for storing each of these entries in the blockstore implementation?